### ISSUE TITLE
Expand default DB beyond Peronosporales to all Oomycetes

### ISSUE THREAD

**Issue by peterjc** [peterjc/thapbi-pict#174]
This would allow us to report matches to the _Pythiales_ (including _Pythium_ and _Phytopythium_), _Saprolegniales_ (including _Achlya_ and _Saprolegnia_) where they are present in the NCBI.

This should make it possible to more usefully compare THAPBI PICT on third party datasets, and might perhaps reveal that a few "unknowns" our nested primers amplified are in fact far more distant cousin _Oomycetes_ than we'd expected.

Current genus breakdown of the default database as of v0.3.10,

```
$ grep -v "^#"  ITS1_DB.txt | cut -f 3 | sort | uniq -c
   2 Basidiophora
 245 Bremia
   2 Calycofera
 299 Hyaloperonospora
  67 Nothophytophthora
   2 Novotelnova
   5 Paraperonospora
  17 Perofascia
  59 Peronosclerospora
 900 Peronospora
13020 Phytophthora
  97 Plasmopara
   2 Plasmoverna
   5 Protobremia
 240 Pseudoperonospora
   4 synthetic
```

**Comment by peterjc** [comment]
From work in progress, increasing from 14966 to 23395 entries in the DB:

```
$ grep -v "^#"  ITS1_DB.txt | cut -f 3 | sort | uniq -c
     17 Achlya
    132 Albugo
    161 Aphanomyces
      1 Aplanopsis
      3 Atkinsiella
      2 Basidiophora
    245 Bremia
      1 Brevilegnia
      2 Calycofera
      1 Diasporangium
      4 Elongisporangium
      1 Geolegnia
    121 Globisporangium
     76 Halophytophthora
    303 Hyaloperonospora
      1 Lagena
    124 Lagenidium
      8 Leptolegnia
      9 Myzocytiopsis
     67 Nothophytophthora
      2 Novotelnova
     14 Ovatisporangium
     33 Paralagenidium
      5 Paraperonospora
     17 Perofascia
     59 Peronosclerospora
    905 Peronospora
      3 Phragmosporangium
  13442 Phytophthora
    339 Phytopythium
      7 Pilasporangium
    189 Plasmopara
      2 Plasmoverna
      3 Plectospira
      5 Protobremia
    240 Pseudoperonospora
     28 Pythiogeton
      9 Pythiopsis
   6476 Pythium
    333 Saprolegnia
      1 Scoliolegnia
      4 synthetic
```

**Comment by peterjc** [comment]
Looking at the QC results on #175, there are *lots* of cases of sequences mapping to completely different genera, e.g.

```
>c61bc4ad1da8630a7c42ebd5828483ab
CCACACCTAAAAATCTTTCCACGTGAATTGTTTTGCTGTACCTTTGGGCTTCGCCGTTGTCTTGTTCTTTTGTAAGAGAAAGGGGGAGGCGCGGTTGGAGGCCATCAGGGGTGTGTTCGTCGCGGTTTGTTTCTTTTGTTGGAACTTGCGCGCGGATGCGTCCTTTTGTCAACCCATTTTTTGAATGAAAAACTGATCATACT
```

Gives:

```
$ thapbi_pict dump | grep c61bc4ad1da8630a7c42ebd5828483ab | cut -f 3 | sort | uniq -c
Wrote 23395 txt format entries
   6 Ovatisporangium
  13 Phytopythium
  17 Pythium
```

This three way split confirmed with an online NCBI BLAST search.

Some could be genuine - others could be taxonomically mislabeled in the NCBI records.

**Comment by peterjc** [comment]
Part of the problem is homotypic synonyms.

e.g. AF271224.1, HQ643955.1, and HQ643954.1 all say *Pythium vexans* - but the NCBI taxonomy considers that and *Ovatisporangium vexans* to be homotypic synonyms of their preferred name *Phytopythium vexans* Taxonomy ID: 907947
 
e.g. AF271230.1, HQ643949.1, HQ643946.1, HQ643948.1 all say *Pythium undulatum* - but the NCBI taxonomy considers that (and others) to be  homotypic synonyms of their preferred name *Phytophthora undulata* taxid 127444.

We have with the *Phytophthora* focus thus far managed to avoid having to implement full synonym support in the DB, see #21.

**Comment by peterjc** [comment]
Done in v0.4.0

**Comment by peterjc** [comment]
v0.4.0 expanded the taxonomy tree to all *Oomycetes* - but not the DB contents.
v0.4.1 added synonym support, which helps with an expanded NCBI import, but issues remain

**Comment by peterjc** [comment]
v0.4.3 added a conflicts command, we don't have any genus conflicts for now - but we're not importing all the NCBI *Oomycetes* yet.

**Comment by peterjc** [comment]
Tried expanding the import today, sadly ``thapbi_pict conflicts`` reports lots of genus-level conflicts:

```
#MD5                             	Level	Conflicts
0215612e8e707ce528d749ea670b75b8	genus	Phytopythium;Pythium
05ca7d9c44096417dd03b1e627c0fc4d	genus	Phytopythium;Pythium
0954a7d6e7ab14015157352af0b45acf	genus	Globisporangium;Pythium
11ffed85d44d3bb625dd115a373e1004	genus	Phytopythium;Pythium
21c7fd14e156d83b255aed0d26c37144	genus	Globisporangium;Pythium
2753d4e07d53d758bfd99ebd2d6c633a	genus	Phytophthora;Pythium
2b949c7d571b8a17391fb4d17311a0fa	genus	Phytopythium;Pythium
312c0be0b4a778e7aa60081d374c82a0	genus	Phytopythium;Pythium
593c97a6e7398518a955ffd767c5a51e	genus	Phytophthora;Phytopythium
63739f92dc4274315ae0fb6d7fcd3d90	genus	Phytophthora;Pythium
7252465de921e972c9d2d88bf3178a58	genus	Phytopythium;Pythium
72e75717b5ba58e9517983093a21c6d6	genus	Phytopythium;Pythium
80f3f9b9fe3e1a46cd5a65514f2281a1	genus	Phytopythium;Pythium
8109213e2d520d8b442f94137cf335ac	genus	Globisporangium;Pythium
8631a93cbbc1dc81ac3cee73becfdec1	genus	Phytopythium;Pythium
87623bdea95623d8c60221f7beed374a	genus	Phytopythium;Pythium
96c839011ca3878680deb99f08126490	genus	Phytopythium;Pythium
a18bef809235c0176937c04d7b94fa88	genus	Phytopythium;Pythium
a3ca8da728374a3a2c2e965fc1f7e9e2	genus	Phytopythium;Pythium
a41bf81910dcd6c54a6338cddefa4b41	genus	Phytopythium;Pythium
a509f07e89e5cd09ed666abe97fae14e	genus	Globisporangium;Pythium
a9128d9aa02ff9b6ca13afc5ea76fc9b	genus	Ovatisporangium;Pythium
b555af21392b86fb598fd40fef207878	genus	Globisporangium;Pythium
c61bc4ad1da8630a7c42ebd5828483ab	genus	Phytopythium;Pythium
c6210b99fb0166d8697d55968506f8f6	genus	Globisporangium;Pythium
cb5da6bfc521fef90fab2b50b8d48421	genus	Hyaloperonospora;Peronospora
d38813d2db59a6a8092b9d2a50bcd324	genus	Globisporangium;Pythium
d3bcf3f104ab546c1c20f78147ece1ef	genus	Globisporangium;Pythium
db2e0825acfd4befa650c19f515f7d34	genus	Phytopythium;Pythium
dca369175ede5c806da9eb3f197a1715	genus	Globisporangium;Pythium
de37e9f5a5d147ba863a06a4ce845da4	genus	Globisporangium;Pythium
eae51dcba19d584bc141de849cfd2a3d	genus	Globisporangium;Pythium
edb50dbe58e820e6e627856acff2ead2	genus	Globisporangium;Pythium
f56971a5bbee9fc0ed6c13106871b9c0	genus	Brevilegnia;Globisporangium;Pythium
f6172ba4cfa0f55804bdcc9dac2550d9	genus	Globisporangium;Pythium
fb647e9842ac70db1137fe8b8df9a638	genus	Phytopythium;Pythium
fc4b476efb9101473c5b026d341a4c22	genus	Globisporangium;Pythium
```

Also, since using v0.5.x, the DB trimming is less strict (no HMM trimming) and we have over 9000 entries in the DB... that might hurt classifier performance?

**Comment by peterjc** [comment]
Given changes in v0.6.x to include the previously discarded leading 32bp region, the pool of NCBI sequences long enough to include this will have dropped quite a bit.

I suspect many of the things ``thapbi_pict ncbi-import`` would load would therefore be missing the start (it currently does not insist on either primer being found and removed).

i.e. We would be adding DB entries which would be no help with the ``identity`` or ``onebp`` classifiers (but might be matched with the fuzzier algorithms).

**Comment by peterjc** [comment]
Closed by #289 and v0.7.7
